### ISSUE TITLE
Request for Review of NIQE Implementation

### ISSUE THREAD

**Issue by songminkyu** [chaofengc/IQA-PyTorch#234]
Hi there,

I am writing to seek your help in reviewing my implementation of the NIQE (Natural Image Quality Evaluator) logic. I have attached my code below and would appreciate it if you could verify whether it is functioning correctly or suggest any improvements.
```python
import pyiqa
import torch
import torch.nn as nn

def calculate_niqe(self, image: np.ndarray) -> Tuple[bool, float]:
    # Convert image to [N, C, H, W] format and normalize to [0, 1] range
    # image: (H, W, C)
    im_t = torch.FloatTensor(image.transpose(2, 0, 1)[None]).to('cuda') / 255.0

    # Create NIQE metric object (using GPU)
    niqe_metric = pyiqa.create_metric("niqe", device='cuda')

    # Calculate NIQE score
    score_tensor = niqe_metric(im_t)
    score = score_tensor.item()

    # Determine quality based on threshold (lower NIQE score means better quality)
    is_good = score < self.niqe_threshold

    return is_good, score

image = cv2.imread(image_path)
isgood, score = calculate_niqe(image)
```
This code is designed to evaluate the quality of an image by calculating its NIQE score. I would particularly like feedback on the following aspects:

Whether the image preprocessing and transformation are handled correctly

If the creation of the NIQE metric object and GPU utilization are implemented properly

The method used for calculating the NIQE score and evaluating image quality

Any feedback you can provide would be greatly appreciated. Thank you for your time and assistance!

**Comment by chaofengc** [comment]
Your implementation appears to be correct! If you have any doubts about your results, you can compare them with the example outputs provided in the following file: [calibration_summary.csv](https://github.com/chaofengc/IQA-PyTorch/blob/main/ResultsCalibra/calibration_summary.csv). The input images can be found in the `ResultsCalibra/dist_dir` directory.
